kartothek/io_components/read.py

```diff
@@ -65,8 +80,11 @@ def dispatch_metapartitions_from_factory(
         assert base_df is not None
         base_df = cast(pd.DataFrame, base_df)
 
-        # Group the resulting MetaParitions by partition keys
-        merged_partitions = base_df.groupby(dataset_factory.partition_keys)
+        # Group the resulting MetaParitions by partition keys or a subset of those keys
+        if dispatch_by:
```

Comment:
At this point you should be able to use `part_keys` as the grouper since you already checked above in which mode we're operating

Reply:
True, thanks.